activesupport/test/deprecation_test.rb

```diff
@@ -256,17 +256,18 @@ def method
   end
 
   def test_deprecate_with_custom_deprecator
-    custom_deprecator = mock('Deprecator') do
-      expects(:deprecation_warning)
-    end
+    Object.const_set :Deprecator, Class.new { def deprecation_warning; end }
```

Comment:
Doesn't this `const_set` on Object muddy the global namespace?

Can't we just do `custom_deprecator = Struct.new(:deprecation_warning).new`?